app/src/main/java/pm/gnosis/heimdall/ui/safe/recover/ReplaceBrowserExtensionActivity.kt

```diff
@@ -0,0 +1,146 @@
+import com.jakewharton.rxbinding2.view.clicks
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.rxkotlin.plusAssign
+import io.reactivex.rxkotlin.subscribeBy
+import kotlinx.android.synthetic.main.layout_replace_browser_extension.*
+import kotlinx.android.synthetic.main.layout_replace_browser_extension_info.*
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.data.repositories.models.SafeTransaction
+import pm.gnosis.heimdall.di.components.ViewComponent
+import pm.gnosis.heimdall.reporting.ScreenId
+import pm.gnosis.heimdall.ui.base.ViewModelActivity
+import pm.gnosis.heimdall.ui.safe.main.SafeMainActivity
+import pm.gnosis.heimdall.utils.errorSnackbar
+import pm.gnosis.model.Solidity
+import pm.gnosis.models.Wei
+import pm.gnosis.svalinn.accounts.base.models.Signature
+import pm.gnosis.svalinn.common.utils.subscribeForResult
+import pm.gnosis.svalinn.common.utils.visible
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+import pm.gnosis.utils.nullOnThrow
+import pm.gnosis.utils.stringWithNoTrailingZeroes
+import timber.log.Timber
+import java.math.BigInteger
+
+class ReplaceBrowserExtensionActivity : ViewModelActivity<ReplaceBrowserExtensionContract>() {
+    private var submissionInProgress = false
+
+    override fun layout() = R.layout.layout_replace_browser_extension
+
+    override fun inject(component: ViewComponent) = viewComponent().inject(this)
+
+    override fun screenId() = ScreenId.REPLACE_BROWSER_EXTENSION_REVIEW
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val txGas = nullOnThrow { intent.getStringExtra(EXTRA_TX_GAS)?.toBigInteger() } ?: run { finish(); return }
+        val dataGas = nullOnThrow { intent.getStringExtra(EXTRA_DATA_GAS)?.toBigInteger() } ?: run { finish(); return }
+        val gasPrice = nullOnThrow { intent.getStringExtra(EXTRA_GAS_PRICE)?.toBigInteger() } ?: run { finish(); return }
+        val safeTransaction = intent.getParcelableExtra<SafeTransaction>(EXTRA_SAFE_TRANSACTION) ?: run { finish(); return }
+        val signature1 =
+            nullOnThrow { intent.getStringExtra(EXTRA_ADDRESS_1).asEthereumAddress()!! to Signature.from(intent.getStringExtra(EXTRA_SIGNATURE_1)) }
+                ?: run { finish(); return }
+        val signature2 =
+            nullOnThrow { intent.getStringExtra(EXTRA_ADDRESS_2).asEthereumAddress()!! to Signature.from(intent.getStringExtra(EXTRA_SIGNATURE_2)) }
+                ?: run { finish(); return }
+
+        val chromeExtensionAddress =
+            nullOnThrow { intent.getStringExtra(EXTRA_CHROME_EXTENSION_ADDRESS).asEthereumAddress()!! } ?: run { finish(); return }
+
+        viewModel.setup(safeTransaction, signature1, signature2, txGas, dataGas, gasPrice, chromeExtensionAddress)
+
+        layout_replace_browser_extension_fee.text = "${viewModel.getMaxTransactionFee().toEther().stringWithNoTrailingZeroes()} ETH"
```

Comment:
don'T we have a string resource for that? If not we should probably add one :D

Reply:
We do have (`R.string.x_ether`). Will fix it.